rusk/src/lib/node/rusk.rs

```diff
@@ -526,6 +527,12 @@ impl Rusk {
     }
 }
 
+async fn finalize_commit(vm: Arc<VM>, commit: [u8; 32]) {
+    if let Err(err) = vm.finalize_commit(commit) {
```

Comment:
as discussed, we should move `vm.finalize_commit` during the state acceptance in `Rusk::accept_transactions`